posthog/hogql_queries/legacy_compatibility/feature_flag.py

```diff
@@ -1,26 +1,39 @@
-from typing import cast
 import posthoganalytics
 from django.conf import settings
-from posthog.cloud_utils import is_cloud
 from posthog.models.user import User
-from django.contrib.auth.models import AnonymousUser
 
+from posthog.schema import InsightType
 
-def hogql_insights_enabled(user: User | AnonymousUser) -> bool:
+
+GLOBAL_FLAG = "hogql-insights-preview"
+INSIGHT_TYPE_TO_FLAG: dict[InsightType, str] = {
+    InsightType.TRENDS: "hogql-insights-trends",
+    InsightType.FUNNELS: "hogql-insights-funnels",
+    InsightType.RETENTION: "hogql-insights-retention",
+    InsightType.PATHS: "hogql-insights-paths",
+    InsightType.LIFECYCLE: "hogql-insights-lifecycle",
+    InsightType.STICKINESS: "hogql-insights-stickiness",
+}
+
+
+def hogql_insights_enabled(user: User, insight_type: InsightType) -> bool:
     if settings.HOGQL_INSIGHTS_OVERRIDE is not None:
         return settings.HOGQL_INSIGHTS_OVERRIDE
 
-    # on PostHog Cloud, use the feature flag
-    if is_cloud():
-        if not hasattr(user, "distinct_id"):  # exclude api endpoints that don't have auth from the flag
-            return False
+    if posthoganalytics.feature_enabled(
+        GLOBAL_FLAG,
+        user.distinct_id,
+        person_properties={"email": user.email},
+        only_evaluate_locally=True,
+        send_feature_flag_events=False,
+    ):
+        # HogQL insights enabled all the way
```

Comment:
I only now understood how this is intended to work 😅 

I was checking if the queries actually use HogQL... so far I don't run into this condition even with all flags enabled, which means I don't get HogQL results. What could be wrong? The flags are enabled and set to release 100%.

Reply:
You should be running this branch with the `hogql-insights-preview` enabled, meaning everything runs on HogQL – definitely works on my machine, hmm

Reply:
I think I double checked... will check again 🏃🏻‍♂️

Reply:
After stepping through with debugger, problem for me is `only_evaluate_locally=True` makes it not get any flag and returns as if it's off for me. When I remove this argument, it works.

Reply:
(need to run locally with a personal API key for local eval to work)
like:
```
POSTHOG_PERSONAL_API_KEY=phx_n79cT52OfsxAWDhZs9j3w67aRoBCZ7l5ksRRKmAi5nr DEBUG=1  ./bin/start
```

Reply:
TIL